ui/src/app/shared/formly/formly-field-modal/formlyfieldmodal.ts

```diff
@@ -1,8 +1,87 @@
+
+import { DataService } from "../../genericComponents/shared/dataservice";
+import { ChannelAddress, Service, Websocket } from "../../shared";
 
 @Component({
     selector: 'formly-field-modal',
-    templateUrl: './formlyfieldmodal.html'
+    templateUrl: './formlyfieldmodal.html',
+    providers: [{
+        useClass: LiveDataService,
+        provide: DataService
+    }]
 })
-export class FormlyFieldModalComponent extends FieldWrapper { }
\ No newline at end of file
+export class FormlyFieldModalComponent extends FieldWrapper implements OnInit, OnDestroy {
+
+    protected formGroup: FormGroup = new FormGroup({});
+    private stopOnDestroy: Subject<void> = new Subject<void>();
+    constructor(
+        protected formBuilder: FormBuilder,
+        protected service: Service,
+        protected websocket: Websocket,
+        protected dataService: DataService
+    ) {
+        super();
+    }
+
+    ngOnInit() {
+        this.service.getCurrentEdge().then(edge => {
+            let channels: ChannelAddress[] = [];
+            let formControls: Map<string, { channel: string, converter?: Function }> = new Map();
+
+            this.props.lines
+                .forEach(line => {
+                    if (line.controlName) {
+                        channels.push(ChannelAddress.fromString(line.channel));
+                        formControls.set(line.controlName, { channel: line.channel, ...(line.converter && { converter: line.converter }) });
+                    }
+                });
+
+
+            if (channels.length === 0) {
+                return;
+            }
+
+            // Prefill formGroup
+            formControls.forEach((channel, key) => {
+                this.formGroup.registerControl(key, new FormControl(null));
+            });
+
+            this.dataService.getValues(channels, edge, this.props.component.id);
+            this.dataService.currentValue.pipe(takeUntil(this.stopOnDestroy), filter(currentData => !!currentData)).subscribe(currentData => {
+                formControls.forEach((control, key) => {
+
+                    // If value for channel equals null or undefined, skip wrong value
+                    if (currentData.allComponents[control.channel] === null || currentData.allComponents[control.channel] === undefined) {
+                        return;
+                    }
+
+                    // If formGroup is dirty, stop overwriting
+                    if (this.formGroup.controls[key]?.dirty) {
+                        return;
+                    }
+
+                    let value: number | string | null;
+                    if (control.converter != null && currentData) {
+                        value = control.converter(currentData);
+                    } else {
+                        value = currentData.allComponents[control.channel];
+                    }
+
+                    this.formGroup.controls[key].setValue(value);
```

Comment:
How would this work for EVCS or does it always assume, that there is a 1-to-1 relation between UI field and Edge config parameter?

Reply:
i have to think about that 🤔

Reply:
i dont know if you like my solution